openedx/core/djangoapps/xblock/runtime/runtime.py

```diff
@@ -226,6 +227,12 @@ def service(self, block, service_name):
         elif service_name == "completion":
             context_key = block.scope_ids.usage_id.context_key
             return CompletionService(user=self.user, context_key=context_key)
+        elif service_name == "user":
+            return DjangoXBlockUserService(
+              self.user,
+              user_is_staff=self.user.is_staff,
+              anonymous_user_id=self.anonymous_student_id,
+            )
```

Comment:
@symbolist Without this change, LabXchange throws `NoSuchServiceError` when viewing CAPA problems:

```
edx.devstack.lms                  | Traceback (most recent call last):
edx.devstack.lms                  |   File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/capa_module.py", line 360, in student_view
edx.devstack.lms                  |     self.lcp
edx.devstack.lms                  |   File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/utils/functional.py", line 80, in __get__
edx.devstack.lms                  |     res = instance.__dict__[self.name] = self.func(instance)
edx.devstack.lms                  |   File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/capa_module.py", line 778, in lcp
edx.devstack.lms                  |     raise LoncapaProblemError(msg).with_traceback(sys.exc_info()[2])
edx.devstack.lms                  |   File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/capa_module.py", line 774, in lcp
edx.devstack.lms                  |     lcp = self.new_lcp(self.get_state_for_lcp())
edx.devstack.lms                  |   File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/capa_module.py", line 810, in new_lcp
edx.devstack.lms                  |     user_service = self.runtime.service(self, 'user')
edx.devstack.lms                  |   File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 2068, in service
edx.devstack.lms                  |     service = self._descriptor_system.service(block, service_name)
edx.devstack.lms                  |   File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/xblock/runtime/runtime.py", line 236, in service
edx.devstack.lms                  |     service = super().service(block, service_name)
edx.devstack.lms                  |   File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/xblock/runtime.py", line 1125, in service
edx.devstack.lms                  |     raise NoSuchServiceError(f"Service {service_name!r} is not available.")
edx.devstack.lms                  | capa.responsetypes.LoncapaProblemError: cannot create LoncapaProblem lb:LabXchange:19cc123d:problem:1-2: Service 'user' is not available.
```

Reply:
[Done](https://github.com/edx/edx-platform/pull/28440/files#diff-be78c43ee5460f0528bc4a888726dafb0f0c207210c3361b535bd34392576b19R233-R234)